docs/manual/working/javaGuide/main/templates/JavaTemplatesDependencyInjection.md

```diff
@@ -0,0 +1,41 @@
+# Dependency Injection with Templates
+
+Twirl templates can now be created with a constructor annotation using `@this`.  The constructor annotation means that Twirl templates can be injected into templates directly and can manage their own dependencies, rather than the controller having to manage dependencies not only for itself, but also for the templates it has to render.
```

Comment:
This makes it sound like `@this` is an annotation like in Java/Scala. I would say something like

> Twirl templates can be generated as a class rather than a static object by declaring a constructor using a special `@this(args)` syntax at the top of the template. This means that Twirl templates can be injected into controllers directly and can manage their own dependencies, rather than the controller having to manage dependencies not only for itself, but also for the templates it has to render.